api/groups/guardianGroup.go

```diff
@@ -148,6 +160,108 @@ func logSignMessage(userIp string, userAgent string, request *requests.SignMessa
 	logArgs = append(logArgs, "error", debugErr.Error())
 }
 
+func (gg *guardianGroup) setSecurityModeNoExpire(c *gin.Context) {
+	var request requests.SetSecurityModeNoExpireMessage
+	var debugErr error
+
+	userIp := c.GetString(mfaMiddleware.UserIpKey)
+	userAgent := c.GetString(mfaMiddleware.UserAgentKey)
+	defer func() {
+		logSetSecurityModeNoExpire(userIp, userAgent, &request, debugErr)
+	}()
+
+	err := json.NewDecoder(c.Request.Body).Decode(&request)
+	if err != nil {
+		debugErr = fmt.Errorf("%w while decoding request", err)
+		returnStatus(c, nil, http.StatusBadRequest, err.Error(), chainApiShared.ReturnCodeRequestError)
+		return
+	}
+	otpCodeVerifyData, err := gg.facade.SetSecurityModeNoExpire(userIp, request)
+	if err != nil {
+		debugErr = fmt.Errorf("%w while setting security mode no expire", err)
+		handleErrorAndReturn(c, getVerifyCodeResponse(otpCodeVerifyData), err.Error())
+		return
+	}
+
+	returnStatus(c, nil, http.StatusOK, "", chainApiShared.ReturnCodeSuccess)
+}
+
+func logSetSecurityModeNoExpire(userIp string, userAgent string, request *requests.SetSecurityModeNoExpireMessage, debugErr error) {
```

Comment:
this is similar with logUnsetSecurityModeNoExpire

suggestion: rename it to `logSecurityModeNoExpire`, add a new string parameter called route, so you can pass either `setSecurityModeNoExpirePath`, either `unsetSecurityModeNoExpirePath`, and reuse it

Reply:
updated